spec.bs

```diff
@@ -2932,6 +2940,202 @@ This specification defines two [=policy-controlled features=] identified by the
+
+NOTE: This is only captured during a [=request=] whose [=request/initiator type=] is `"fetch"`, made
+with the {{RequestInit/adAuctionHeaders}} option set to `true`, as described in the
+[:Ad-Auction-Signals:] header description.
+
+<dl dfn-for="direct from seller signals">
+: <dfn>auction signals</dfn>
+:: Null or a [=string=].
+  Opaque JSON data passed to both buyers' and the seller's [=script runners=].
+: <dfn>seller signals</dfn>
+:: Null or a [=string=].
+  Opaque JSON data passed to the seller's [=script runner=].
+: <dfn>per buyer signals</dfn>
+:: A [=map=] whose [=map/keys=] are [=origins=] and whose [=map/values=] are [=strings=].
+  [=map/Keys=] are buyers and must be valid HTTPS origins. [=map/Values=] are opaque JSON data
+  passed to corresponding buyer's [=script runner=].
+
+</dl>
+
+Each [=traversable navigable=] has a <dfn for="traversable navigable">captured ad auction headers
+</dfn>, which is a [=map=] whose [=map/keys=] are [=direct from seller signals keys=] and whose
+[=map/values=] are [=direct from seller signals=].
+</div>
+
+<div algorithm="fetch auction headers patch">
+The following will be added to [[Fetch#http-extensions]]:
+
+<h3 id=sec-ad-auction-fetch-header>The \`<a http-header><code>Sec-Ad-Auction-Fetch</code></a>\` HTTP
+request header</h3>
+
+The \`<dfn http-header><code>Sec-Ad-Auction-Fetch</code></dfn>\` request header is an optional
+[=structured header=] with of type [=structured header/boolean=]. [:Sec-Ad-Auction-Fetch:] will only
+be set on a [=request=] whose [=request/initiator type=] is `"fetch"`, made with the
+{{RequestInit/adAuctionHeaders}} option set to `true`. If [:Sec-Ad-Auction-Fetch:] is equal to `?1`,
+the user agent will remove any [:Ad-Auction-Signals:] from the returned [=response=] -- the
+[:Ad-Auction-Signals:] value will instead only be used in Protected Audiences auctions. 
+
+<h3 id=ad-auction-signals-header>The \`<a http-header><code>Ad-Auction-Signals</code></a>\` HTTP
+response header</h3>
+
+The \`<dfn http-header><code>Ad-Auction-Signals</code></dfn>\` response header provides value
+of a JSON array of dictionaries, each with an `adSlot` key. Protected Audience non-component,
+component, and top-level auctions may specify which signals to load by the `adSlot` key.
+</div>
+
+<div algorithm="fetch auction signals redirect patch">
+The following step will be added to the [=HTTP fetch=] algorithm, immediately under the step "If
+<var ignore>internalResponse</var>’s [=status=] is a [=redirect status=]:"
+
+1. [=header list/Delete=] "[:Ad-Auction-Signals:]" from <var ignore>response</var>'s
+  [=response/header list=].
+
+</div>
+
+<div algorithm="fetch auction signals response patch">
+The following step will be added to the [=HTTP fetch=] algorithm, before step
+"Return |response|.":
+
+1. If |response| is not null, |response|'s [=status=] is not a [=redirect status=], |fetchParams|'s

```

Comment:
Note that `|response|` can also be a network error. Should we just only run this step if `|response|` has an OK status: https://fetch.spec.whatwg.org/#ok-status? Does that match the implementation?

Reply:
It looks like network errors have an empty header list by default, so the algorithm that tries to pluck the relevant ad auction headers off of the header list (and handles the empty-headers case appropriately) would just work by default, so maybe we don't need special error handling here after all I think?

Reply:
Yep -- and also confirmed that [0] does get called for thinks like HTTP 404.

[0] https://source.chromium.org/chromium/chromium/src/+/main:content/browser/interest_group/ad_auction_url_loader_interceptor.cc;l=127;drc=17c414554403dabe9caca7ec356f564b46ebc794